unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx

```diff
@@ -90,11 +98,39 @@ export const LocksForm = ({
   const [defaultValue, setDefaultValue] = useState<Record<string, any>>({})
   const [recurring, setRecurring] = useState<string | number>('')
   const [recurringUnlimited, setRecurringUnlimited] = useState(false)
+  const [lockRecurring, setLockRecurring] = useState<RecurringByLock>({})
 
   const { getIsRecurringPossible } = useLockSettings()
 
   const [locks, setLocks] = useState<LocksProps>(locksDefault)
 
+  // preload default and set recurring (es. saved config)
+  useEffect(() => {
+    const getRecurringCb = async () => {
```

Comment:
Please use a better function name

Reply:
Also I think I have already seen you write code to check if a lock is recurring somewhere? Are you sure your code is not duplicated?

Reply:
We need both: 
- when we add a new lock to the list, check if recurring is possible 
- for an existing saved config with locks, we fetch with the code above